### ISSUE TITLE
Rule test harness check for duplicated test cases incorrectly flags those with unicode/non-unicode characters

### ISSUE THREAD

**Issue by scalvert** [ember-template-lint/ember-template-lint#2329]
The changes in https://github.com/ember-template-lint/ember-template-lint/pull/2279 have caused a slight regression, where previously the following was allowed:

```js
  good: [
    `{{t-def "you\u2019re closer than you think" key="foo"}}`,
    `{{t-def "you’re closer than you think" key="foo"}}`,
    `{{t-def "she said \u2018hello\u2019" key="foo"}}`,
    `{{t-def "she said ‘hello’" key="foo"}}`,
    `{{t-def "hello\u2026" key="foo"}}`,
    `{{t-def "hello…" key="foo"}}`,
  ],
```

Now these are being flagged as duplicates, because when we `JSON.stringify` the test cases, it converts unicode characters to string equivalents.

**Comment by bmish** [comment]
Good catch. I need to figure out a fix for this.

**Comment by scalvert** [comment]
I wonder if this is even something we should care about, and whether the serialization is fine as it is. I say this because

```shell
> `\u2019` === `’`
true
```

because both represent the same code point. So, perhaps this is accurately flagging these as redundant test cases?

**Comment by bmish** [comment]
Good point....if the resulting string template values that are being tested are identical, then these would be actual duplicates.

**Comment by scalvert** [comment]
I think we should keep this as is, unless anyone else can provide a reasonable justification for trying to accommodate it. As is, this seems to work as intended. I'm going to close this and remove the offending test cases.

Thanks, @bmish!
